libraries/lib-cloud-audiocom/UploadService.cpp

```diff
@@ -0,0 +1,475 @@
+/*  SPDX-License-Identifier: GPL-2.0-or-later */
+/*!********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  OAuthService.cpp
+
+  Dmitry Vedenko
+
+**********************************************************************/
+
+#include "UploadService.h"
+
+#include <mutex>
+
+#include <wx/filefn.h>
+#include <wx/filename.h>
+
+#include <rapidjson/document.h>
+#include <rapidjson/writer.h>
+
+#include "OAuthService.h"
+#include "ServiceConfig.h"
+
+#include "NetworkManager.h"
+#include "Request.h"
+#include "IResponse.h"
+#include "MultipartData.h"
+
+#include "CodeConversions.h"
+
+namespace cloud::audiocom
+{
+namespace
+{
+std::string_view DeduceMimeType(const wxString& ext)
+{
+   if (ext == "wv")
+      return "audio/x-wavpack";
+   else if (ext == "flac")
+      return "audio/x-flac";
+   else if (ext == "mp3")
+      return "audio/mpeg";
+   else
+      return "audio/x-wav";
```

Comment:
Four types recognized here, but only three in the enumeration in ServiceConfig.h?
I notice now, that enum is only for the return of `AudioFormat GetPreferredAudioFormat` and it's always the same.  Did you forget to define an `EnumSetting` to save it as a preference?